### ISSUE TITLE
updator does not delete the zip file it downloads

### ISSUE THREAD

**Issue by nickgaw** [munawarb/Three-D-Velocity#94]
I noticed that when the game updates itself the updator says it is cleaning up but there is still the zip file it downloads left behind in the directory that all the binaries are in.  Also, if you just wish to run the updator to see if there is a new version of the game and you run the updator binary it errors out.  I think it would be nice if there were a command line option to just check for updates and update and then quit.  The reason I ask about an update checking method as currently after the game starts and before it gets to the main menu the game crashes and I have attached the error log for this to this bug report.

**Comment by munawarb** [comment]
Hi @nickgaw,
It doesn't look like the error log got posted. Can you reupload it?

**Comment by nickgaw** [comment]
I tried to attach it but have to paste it into the edit box as for some reason github won't upload the log file.
Error log, created with build version 2.31.0.0: 
Error base exception: SharpDX.SharpDXException: HRESULT: [0x88960001], Module: [SharpDX.XAudio2], ApiCode: [XAUDIO2_E_INVALID_CALL/InvalidCall], Message: Unknown
   at SharpDX.Result.CheckError()
   at SharpDX.XAudio2.Voice.SetVolume(Single volume, Int32 operationSet)
   at BPCSharedComponent.ExtendedAudio.OggBuffer.process(Int32 playPointer, Boolean loop) in C:\Users\munawar\projects\Three-D-Velocity\shared\BPCSharedComponent\OggBuffer.cs:line 113
   at BPCSharedComponent.ExtendedAudio.OggBuffer.play(Boolean loop) in C:\Users\munawar\projects\Three-D-Velocity\shared\BPCSharedComponent\OggBuffer.cs:line 88
   at TDV.GUI.mainMenu() in C:\Users\munawar\projects\Three-D-Velocity\TDV\GUI.cs:line 403
   at TDV.GUI.startGame() in C:\Users\munawar\projects\Three-D-Velocity\TDV\GUI.cs:line 356
Error Description: HRESULT: [0x88960001], Module: [SharpDX.XAudio2], ApiCode: [XAUDIO2_E_INVALID_CALL/InvalidCall], Message: Unknown
Stack trace:    at SharpDX.Result.CheckError()
   at SharpDX.XAudio2.Voice.SetVolume(Single volume, Int32 operationSet)
   at BPCSharedComponent.ExtendedAudio.OggBuffer.process(Int32 playPointer, Boolean loop) in C:\Users\munawar\projects\Three-D-Velocity\shared\BPCSharedComponent\OggBuffer.cs:line 113
   at BPCSharedComponent.ExtendedAudio.OggBuffer.play(Boolean loop) in C:\Users\munawar\projects\Three-D-Velocity\shared\BPCSharedComponent\OggBuffer.cs:line 88
   at TDV.GUI.mainMenu() in C:\Users\munawar\projects\Three-D-Velocity\TDV\GUI.cs:line 403
   at TDV.GUI.startGame() in C:\Users\munawar\projects\Three-D-Velocity\TDV\GUI.cs:line 356

**Comment by nickgaw** [comment]
I redownloaded the binary zip file and ran it again with my saved missions and when the fighters come at me when going to the power plant this crash happens.  I made sure to rerun the directX setup to make sure everything was installed properly.
Error log, created with build version 2.31.0.0: 
Error base exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at TDV.Mission.createNewFighter(Int32 type, Int32 radius) in C:\Users\munawar\projects\Three-D-Velocity\TDV\Mission.cs:line 501
   at TDV.Aircraft.startFighterSwarm() in C:\Users\munawar\projects\Three-D-Velocity\TDV\Aircraft.cs:line 4266
   at TDV.Aircraft.move() in C:\Users\munawar\projects\Three-D-Velocity\TDV\Aircraft.cs:line 849
   at TDV.Holder.activate() in C:\Users\munawar\projects\Three-D-Velocity\TDV\Holder.cs:line 106
Error Description: Object reference not set to an instance of an object.
Stack trace:    at TDV.Mission.createNewFighter(Int32 type, Int32 radius) in C:\Users\munawar\projects\Three-D-Velocity\TDV\Mission.cs:line 501
   at TDV.Aircraft.startFighterSwarm() in C:\Users\munawar\projects\Three-D-Velocity\TDV\Aircraft.cs:line 4266
   at TDV.Aircraft.move() in C:\Users\munawar\projects\Three-D-Velocity\TDV\Aircraft.cs:line 849
   at TDV.Holder.activate() in C:\Users\munawar\projects\Three-D-Velocity\TDV\Holder.cs:line 106
